enterprise/frontend/src/embedding-sdk/types/plugins.ts

```diff
@@ -0,0 +1,52 @@
+import type { ReactNode } from "react";
+
+import type { DashCardMenuItem } from "./dashboard";
+import type { Question } from "./question";
+
+export type MetabaseClickAction = {
+  name: string;
+} & Record<string, any>;
+
+export type MetabaseDataPointObject = {
+  value?: string | number | null | boolean;
+  column?: Record<string, any>;
+  event?: MouseEvent;
+  data?: {
+    col: Record<string, any> | null;
+    value: string | number | null | boolean;
+  }[];
+};
+
+export type MetabaseClickActionPluginsConfig = (
+  clickActions: MetabaseClickAction[],
+  clickedDataPoint: MetabaseDataPointObject,
+) => MetabaseClickAction[];
+
+export type DashboardCardMenuCustomElement = ({
+  question,
+}: {
+  question: Question;
+}) => ReactNode;
+
+export type CustomDashboardCardMenuItem = ({
+  question,
+}: {
+  question?: Question;
+}) => DashCardMenuItem;
+
+export type DashboardCardCustomMenuItem = {
+  withDownloads?: boolean;
+  withEditLink?: boolean;
+  customItems?: (DashCardMenuItem | CustomDashboardCardMenuItem)[];
+};
+
+export type MetabaseDashboardPluginsConfig = {
+  dashboardCardMenu?:
+    | DashboardCardMenuCustomElement
+    | DashboardCardCustomMenuItem;
+};
+
+export type MetabasePluginsConfig = {
+  mapQuestionClickActions?: MetabaseClickActionPluginsConfig;
+  dashboard?: MetabaseDashboardPluginsConfig;
+};
```

Comment:
This file almost fully copies the internal `frontend/src/metabase/embedding-sdk/types/plugins.ts` file except more simple versions of `MetabaseClickAction` and `MetabaseDataPointObject` that dont reference by a chain a half of metabase types

Reply:
Could we replace the internal `frontend/src/metabase/embedding-sdk/types/plugins.ts` with the type from this file? Or does it break our internal types? That way we can share the same `MetabasePluginsConfig` without having an internal version.

Reply:
We should not reference any internal types in types that we export from SDK.
It means that all plugins props should reference MetabasePluginsConfig that does not import anything from frontend/src/metabase
That's why we have 2 plugins files now and 2 types for question - one is public, one is internal (edited)